src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -55,9 +55,9 @@ private function locateFilesInFilesInFilesDefinitions(array $locations) : Genera
         foreach ($locations as $location) {
             if (is_file($location)) {
                 yield $location;
+            } elseif (is_dir($location)) {
+                yield from $this->extractFilesFromDirectory($location);
             }
-
-            yield from $this->extractFilesFromDirectory($location);
```

Comment:
...so I think there aren't any alternatives. I'd still leave the `elseif`in though, and not change it to 
```php
if (is_file($location)) {
    yield $location;
    continue;
}
yield from $this->extractFilesFromDirectory($location);
```
if you were implying that.